babel/babel
# Remove the sourceMapTarget option from core and implement it in babel-cli.

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          | N
| Major: Breaking Change?  | Y
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR         | 
| Any Dependency Changes?  |
| License                  | MIT

Ok, background first, the `sourceMapTarget` option was passed straight through `babel-core` into the `file` field of the sourcemap created for the output. The `file` field is optional in sourcemaps, but up until now we've actually had it default to the _input_ filename, but per the spec it should really be the output file name.

The issue is for most cases, you can't know whether the filename should end with .js or .mjs. That leaves us a few options:

1. Allow it to be a function that returns the output filename
2. Make it default to undefined, so users can choose to set it if they want, but otherwise it defaults to empty.
3. Remove it entirely and let the caller set it if they want once they get the sourcemap back.

(1) sounds like a painful API, so I vetoed that.
(2) is an option, but I started to question the necessity for this option to begin with. If a sourcemap is inlined into the output as a comment, including the filename doesn't really make sense. If it _isn't_ inlined into the file, then it is trivial for the tooling that calls Babel to assign `.file` on the map itself afterward, rendering the option useless.

So essentially all that this PR removes is the ability to set the `file` field on inline sourcemap JSON.

If anyone feel that matters, I'd be happy to go with option (2) instead.
